pyproject_metadata/__init__.py

```diff
@@ -405,10 +423,52 @@ def validate(self) -> None:
             )
             raise ConfigurationError(msg)
 
+        if self.license_files is not None and isinstance(self.license, License):
+            msg = '"project.license-files" must not be used when "project.license" is not a SPDX license expression'
+            raise ConfigurationError(msg)
+
+        if isinstance(self.license, str) and any(
+            c.startswith('License ::') for c in self.classifiers
+        ):
+            msg = 'Setting "project.license" to an SPDX license expression is not compatible with "License ::" classifiers'
+            raise ConfigurationError(msg)
+
+        if show_warnings and self.metadata_version not in PRE_SPDX_METADATA_VERSIONS:
+            if isinstance(self.license, License):
+                warnings.warn(
+                    'Set "project.license" to an SPDX license expression for metadata >= 2.4',
+                    ConfigurationWarning,
+                    stacklevel=2,
+                )
+            elif any(c.startswith('License ::') for c in self.classifiers):
+                warnings.warn(
+                    '"License :: * classifiers" are deprecated for metadata >= 2.4, use a SPDX license expression for "project.license" instead',
```

Comment:
Quotes in the wrong place? Also I interpret the classifiers as namespaced, thus the `*` can maybe be omitted:
```suggestion
                    '"License ::" classifiers are deprecated for metadata >= 2.4, use a SPDX license expression for "project.license" instead',
```

Reply:
Also, unless I'm missing something, I don't see how this warning is reachable. If `project.license` is set to a table, the previous warning is emitted, thus the `elif` branch is not taken. And if the it is set to a SPDX license expression, having the `License ::` classifiers is an hard error emitted a few lines above.

Reply:
Oh, it is emitted if `project.license` is not set and the classifiers are present. I think it makes sense.

Reply:
Yes, this was in the "tools MAY warn" category.